File: bulk/java/TempFile01.java
// CWE-377: Insecure Temporary File creation
import java.io.*;
public class TempFile01 {
    public File createTemp(String prefix) throws IOException {
        File f = new File("/tmp/" + prefix + System.currentTimeMillis());
        f.createNewFile();
        return f;
    }
}

File: bulk/go/tempfile_01.go
// CWE-377: Insecure temp file with predictable name
package main
import (
	"fmt"
	"os"
)
func writeTmp(data []byte) (*os.File, error) {
	f, err := os.CreateTemp("/tmp", "upload_*")
	if err != nil {
		return nil, err
	}
	return f, nil
}
